src/coordinator/shard_map.hpp

```diff
@@ -115,7 +118,12 @@ struct ShardMap {
     bool mutated = false;
 
     for (auto &[label_id, label_space] : label_spaces) {
-      for (auto &[low_key, shard] : label_space.shards) {
+      for (auto it = label_space.shards.begin(); it != label_space.shards.end(); it++) {
+        auto &[low_key, shard] = *it;
+        std::optional<PrimaryKey> high_key;
+        if (const auto next_it = std::next(it); next_it != label_space.shards.end()) {
+          high_key = next_it->first;
+        }
```

Comment:
This is important, because without the max key the shards cannot determine properly whether a vertex should be stored on them or not. This functionality is necessary when creating edges.